src/main/scala/io/iohk/ethereum/blockchain/sync/SyncBlockBodiesRequestHandler.scala

```diff
@@ -0,0 +1,46 @@
+package io.iohk.ethereum.blockchain.sync
+
+import akka.actor.{ActorRef, Props, Scheduler}
+import akka.util.ByteString
+import io.iohk.ethereum.blockchain.sync.SyncController.BlockBodiesReceived
+import io.iohk.ethereum.db.storage.AppStateStorage
+import io.iohk.ethereum.network.p2p.messages.PV62.{BlockBodies, GetBlockBodies}
+
+class SyncBlockBodiesRequestHandler(
+    peer: ActorRef,
+    requestedHashes: Seq[ByteString],
+    appStateStorage: AppStateStorage)(implicit scheduler: Scheduler)
+  extends FastSyncRequestHandler[GetBlockBodies, BlockBodies](peer) {
+
+  override val requestMsg = GetBlockBodies(requestedHashes)
+  override val responseMsgCode: Int = BlockBodies.code
+
+  override def handleResponseMsg(blockBodies: BlockBodies): Unit = {
+    if (blockBodies.bodies.isEmpty) {
+      fastSyncController ! BlacklistSupport.BlacklistPeer(peer)
```

Comment:
https://github.com/input-output-hk/etc-client/pull/99 Lets discuss how to proceed.